### PR TITLE
Switch to image/next.js

### LINKED ISSUE #86: Migrate to next/image
Use [this](https://nextjs.org/docs/api-reference/next/image#src) documentation to migrate to next.js Image component. This will make your images load much smarter.

Note: you'll need to convert relative URLs to absolute URLs (eg: `/Data/Media/posts/1.jpg` to `http://localhost:3000/Data/Media/posts/1.jpg`). To do that, you can do

```js
const fullURL = new URL(relativeURL, window.location.href).href
```

This means, please resolve `relativeURL` against the current URL (`window.location.href`) and give me an absolute URL.

### PR DESCRIPTION
**Pull request by alshakerM**
Switched to image/next.js.

Fixes #86

### CODE REVIEW

File: components/pages/Messages/Messages.js [hunk 1]
```diff
@@ -188,11 +195,15 @@ export function Messages({ fromUserId }) {
...
                           threadData?.messages[nextMessage]?.direction &&
                         message.direction !== 'sent' ? (
-                          <img
-                            src={threadData?.from_user_thumbnail}
-                            alt="sender profile pic"
-                            className={styles.messagesSenderAvatar}
-                          />
+                          threadData.from_user_id && (
+                            <Image
+                              src={threadData?.from_user_thumbnail}
```
Comment (alshakero) [inline components/pages/Messages/Messages.js:200]: We know threadData is not null here.
```suggestion
                              src={threadData.from_user_thumbnail}
```

File: components/pages/Messages/Messages.module.css [hunk 2]
```diff
@@ -62,8 +62,6 @@
   background-color: #fafafa;
 }
 .senderThumbnail {
-  height: 56px;
-  width: 56px;
```
Comment (alshakero) [inline components/pages/Messages/Messages.module.css:66]: These should stay.

File: components/pages/Messages/Messages.module.css [hunk 3]
```diff
@@ -93,8 +91,6 @@
   box-sizing: border-box;
 }
 .messagesSenderAvatar {
-  height: 24px;
-  width: 24px;
```
Comment (alshakero) [inline components/pages/Messages/Messages.module.css:97]: These too.

File: next.config.js [hunk 4]
```diff
@@ -1,12 +1,5 @@
...
-    return [
-      {
-        source: '/direct',
-        destination: '/direct/inbox',
-        permanent: true,
-      },
-    ];
+  images: {
+    domains: ['picsum.photos'],
   },
```
Comment (alshakero) [inline next.config.js:4]: Why delete these? They're needed to redirect from /direct to /direct/inbox.

**Review by alshakero — CHANGES_REQUESTED** [review]
Clean! But for `Image`. The `width` and `height` are the width and height of the jpg file itself, not for how you'd like to appear. So keep sizes in CSS and change `width` and `height` values to the JPG's.

**Comment by alshakerM** [comment]
Changed the width and height of Next.js/Image to be the original pic width and height.
Added a parent div to the Next.js/Image to add the needed size for the Image

File: components/Post/Post.js [hunk 5]
```diff
@@ -64,7 +63,7 @@ function PostActions({ postId, is_post_liked, isExpanded }) {
       <div className={styles.likeShareTelegramIcons}>
         <LikeButton
           is_post_liked={is_post_liked}
-          onClick={() => togglePostLike(postId)}
+          onClick={() => postLike(postId, !is_post_liked)}
```
Comment (alshakero) [inline components/Post/Post.js:66]: This is out of scope. Let's leave it this time, but next time please create another tiny PR for such things.

File: components/Post/Post.js [hunk 6]
```diff
@@ -621,7 +626,7 @@ export function Post({ isIndependentPost, datum, index, isFloating }) {
           />
 
           <PostDate posting_time={datum.posting_time} isExpanded={isExpanded} />
-          <PostInput index={index} isExpanded={isExpanded} />
+          <PostInput index={datum.post_id} isExpanded={isExpanded} />
```
Comment (alshakero) [inline components/Post/Post.js:629]: Same.

File: components/PostImage/PostImage.js [hunk 7]
```diff
@@ -1,6 +1,29 @@
...
+    }
+    return () => {
+      if (imgRef.current && isVisible) {
+        observer.disconnect(imgRef.current);
+      }
+    };
+  }, [imgRef, options]);
+
+  return [imgRef, isVisible];
+};
```
Comment (alshakero) [inline components/PostImage/PostImage.js:26]: This doesn't belong here 🤔

File: components/PostImage/PostImage.js [hunk 8]
```diff
@@ -9,8 +32,23 @@ export function PostImage({
...
+    root: null,
+    rootMargin: '0px',
+    threshold: 0,
+  });
+
+  React.useEffect(() => {
+    if (!isActive) {
+      setIsImgCLicked(false);
+    }
+  }, [isActive]);
```
Comment (alshakero) [inline components/PostImage/PostImage.js:51]: Why are these here? :\

File: components/PostImage/PostImage.js [hunk 9]
```diff
@@ -50,6 +92,58 @@ export function PostImage({
...
+            fill="#ffffff"
+            height="12"
+            role="img"
+            viewBox="0 0 24 24"
+            width="12"
+          >
+            <path d="M21.334 23H2.666a1 1 0 01-1-1v-1.354a6.279 6.279 0 016.272-6.272h8.124a6.279 6.279 0 016.271 6.271V22a1 1 0 01-1 1zM12 13.269a6 6 0 116-6 6.007 6.007 0 01-6 6z"></path>
+          </svg>
+        </div>
+      )}
```
Comment (alshakero) [inline components/PostImage/PostImage.js:146]: The whole tags PR seems here.

File: components/pages/UserProfile/UserProfile.module.css [hunk 10]
```diff
@@ -162,9 +162,10 @@
   display: flex;
 }
 .postsSection {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  grid-template-columns: 293px 293px 293px;
```
Comment (alshakero) [inline components/pages/UserProfile/UserProfile.module.css:166]: You can change it back to flex so it works on mobiles.

File: mocking-scripts/add-tags-to-media.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+        ];
+        return photo;
+      });
+    post.post_has_tags = true;
+  }
+  post.post_has_tags = false;
+  return post;
+});
+
+writeFileSync('../server/posts-data.json', JSON.stringify(posts));
```
Comment (alshakero) [inline mocking-scripts/add-tags-to-media.js:31]: Tags PR.

File: pages/api/posts.js [hunk 12]
```diff
@@ -1,25 +1,113 @@
...
-  if (postId) {
-    const post = posts.find((p) => p.post_id === postId);
-    if (post) {
-      res.status(200).json(post);
-    } else {
-      res.status(404).send('Not found.');
+  if (req.method === 'POST') {
+    const { action } = req.query;
+    const { postId, like, commentId, text, replyId } = req.body;
+    console.log('omar1', { postId, like, commentId, text, replyId });
```
Comment (alshakero) [inline pages/api/posts.js:21]:
```suggestion
```

File: pages/api/posts.js [hunk 13]
```diff
@@ -1,25 +1,113 @@
...
+          reply.is_liked_by_user = !reply.is_liked_by_user;
+          syncPostFile();
+          return res.status(200).json({
+            ok: true,
+            message: `reply is now ${
+              reply.is_liked_by_user ? 'liked' : 'unliked'
+            }`,
+          });
+        } catch (e) {
+          console.log('omar', e);
```
Comment (alshakero) [inline pages/api/posts.js:70]:
```suggestion

```

File: stores/messagesStore/controls.js [hunk 14]
```diff
@@ -13,3 +13,42 @@ export async function REQUEST_THREADS(action) {
...
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      fromUserId: action.fromUserId,
+      text: action.text,
+    }),
+  });
+}
```
Comment (alshakero) [inline stores/messagesStore/controls.js:54]: I can't explain this. I don't know how this happened.

**Review by alshakero — CHANGES_REQUESTED** [review]
This PR has the changes of the last 3 PRs together.

File: components/Post/Post.js (hunk 5 above)
Comment (alshakerM) [reply components/Post/Post.js:66]: I think this is from the rebase.

File: components/Post/Post.js (hunk 6 above)
Comment (alshakerM) [reply components/Post/Post.js:629]: + 1

File: stores/messagesStore/controls.js (hunk 14 above)
Comment (alshakerM) [reply stores/messagesStore/controls.js:54]: Me either

File: pages/api/posts.js (hunk 13 above)
Comment (alshakerM) [reply pages/api/posts.js]: Addressed 1 to 1.

File: pages/api/posts.js (hunk 12 above)
Comment (alshakerM) [reply pages/api/posts.js]: Addressed 1 to 1.

File: mocking-scripts/add-tags-to-media.js (hunk 11 above)
Comment (alshakerM) [reply mocking-scripts/add-tags-to-media.js]: Addressed 1 to 1.

File: components/pages/UserProfile/UserProfile.module.css (hunk 10 above)
Comment (alshakerM) [reply components/pages/UserProfile/UserProfile.module.css]: Done.

File: components/PostImage/PostImage.js (hunk 9 above)
Comment (alshakerM) [reply components/PostImage/PostImage.js]: Addressed 1 to 1.

File: components/PostImage/PostImage.js (hunk 8 above)
Comment (alshakerM) [reply components/PostImage/PostImage.js]: Addressed 1 to 1.

**Review by alshakero — APPROVED** [review]
LGTM now. The issue was fixed by a rebase.

File: components/PostImage/PostImage.js (hunk 7 above)
Comment (alshakerM) [reply components/PostImage/PostImage.js]: Addressed 1 to 1.
